// @vuepress/bundler-webpack/lib/dev/createDevServerConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDevServerConfig = void 0;
const path_1 = require("path");
const utils_1 = require("@vuepress/utils");
const trailingSlashMiddleware_1 = require("./trailingSlashMiddleware");
const createDevServerConfig = (app, options) => {
    // TODO: add types for webpack-dev-server 4
    const serverConfig = {
        compress: true,
        devMiddleware: {
            publicPath: app.options.base,
            writeToDisk: false,
            stats: app.env.isDebug ? 'normal' : 'errors-warnings',
        },
        firewall: false,
        headers: {
            'access-control-allow-origin': '*',
        },
        historyApiFallback: {
            disableDotRule: true,
            rewrites: [{ from: /./, to: utils_1.path.join(app.options.base, 'index.html') }],
        },
        host: app.options.host,
        hot: true,
        onAfterSetupMiddleware: (expressApp, server) => {
            var _a;
            // plugin hook: afterDevServer
            (_a = options.afterDevServer) === null || _a === void 0 ? void 0 : _a.call(options, expressApp, server);
        },
        onBeforeSetupMiddleware: (expressApp, server) => {
            var _a;
            // use trailing slash middleware to support vuepress routing in dev-server
            // it will be handled by most of the deployment platforms
            expressApp.use(trailingSlashMiddleware_1.trailingSlashMiddleware);
            // plugin hook: beforeDevServer
            (_a = options.beforeDevServer) === null || _a === void 0 ? void 0 : _a.call(options, expressApp, server);
        },
        open: app.options.open,
        port: app.options.port,
        static: {
            // `static.directory` will fail on Windows if we do not replace / with \
            directory: app.dir.public().replace('/', path_1.sep),
            publicPath: app.options.base,
            watch: {
                ignoreInitial: true,
                ignored: [
                    // Do not watch node_modules
                    'node_modules',
                ],
            },
        },
    };
    return serverConfig;
};
exports.createDevServerConfig = createDevServerConfig;
