// @vuepress/cli/lib/utils/esbuildUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildToCodeSync = exports.buildToCode = exports.createBuildToCodeOptions = void 0;
const esbuild_1 = require("esbuild");
/**
 * Create esbuild `BuildOptions` for `buildToCode`
 */
const createBuildToCodeOptions = (filePath) => ({
    entryPoints: [filePath],
    format: 'cjs',
    platform: 'node',
    target: 'node12',
    bundle: true,
    write: false,
    external: ['fsevents', 'vuepress', '@vuepress/*'],
});
exports.createBuildToCodeOptions = createBuildToCodeOptions;
/**
 * Take a file as entry point, and build it to cjs code
 */
const buildToCode = async (filePath) => {
    const buildResult = await esbuild_1.build(exports.createBuildToCodeOptions(filePath));
    return buildResult.outputFiles[0].text;
};
exports.buildToCode = buildToCode;
/**
 * Sync version fo `buildToCode`
 */
const buildToCodeSync = (filePath) => {
    const buildResult = esbuild_1.buildSync(exports.createBuildToCodeOptions(filePath));
    return buildResult.outputFiles[0].text;
};
exports.buildToCodeSync = buildToCodeSync;
